bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ChannelI18nUtil.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.thing.i18n;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Function;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.type.ChannelDefinition;
+import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeRegistry;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.osgi.framework.Bundle;
+
+/**
+ * A utility service which localises {@link ChannelDefinition}.
+ * Falls back to a localised {@link ChannelType} for label and description when not given otherwise.
+ *
+ * @see {@link ThingTypeI18nLocalizationService}
+ * @see {@link ChannelGroupTypeI18nLocalizationService}
+ *
+ * @author Henning Treu - initial contribution
+ *
+ */
+@NonNullByDefault
+class ChannelI18nUtil {
```

Comment:
Instead of defining it package private, wouldn't the proper way be to move it to an internal package?